### ISSUE TITLE
Saving llama3 finetuned checkpoint saving fails (deepspeed, multi GPU)

### ISSUE THREAD

**Issue by mariokostelac** [axolotl-ai-cloud/axolotl#1617]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

After training with deepspeed (8GPUs) succeeds, saving the checkpoint should succeed too. 

### Current behaviour

It seems that model saving happens from all ranks so one of them succeeds and the rest fails with 

```
[rank4]: Traceback (most recent call last):
[rank4]:   File "<frozen runpy>", line 198, in _run_module_as_main
[rank4]:   File "<frozen runpy>", line 88, in _run_code
[rank4]:   File "/home/ec2-user/axolotl/src/axolotl/cli/train.py", line 59, in <module>
[rank4]:     fire.Fire(do_cli)
[rank4]:   File "/home/ec2-user/micromamba/envs/axolotl/lib/python3.11/site-packages/fire/core.py", line 143, in Fire
[rank4]:     component_trace = _Fire(component, args, parsed_flag_args, context, name)
[rank4]:                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank4]:   File "/home/ec2-user/micromamba/envs/axolotl/lib/python3.11/site-packages/fire/core.py", line 477, in _Fire
[rank4]:     component, remaining_args = _CallAndUpdateTrace(
[rank4]:                                 ^^^^^^^^^^^^^^^^^^^^
[rank4]:   File "/home/ec2-user/micromamba/envs/axolotl/lib/python3.11/site-packages/fire/core.py", line 693, in _CallAndUpdateTrace
[rank4]:     component = fn(*varargs, **kwargs)
[rank4]:                 ^^^^^^^^^^^^^^^^^^^^^^
[rank4]:   File "/home/ec2-user/axolotl/src/axolotl/cli/train.py", line 35, in do_cli
[rank4]:     return do_train(parsed_cfg, parsed_cli_args)
[rank4]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank4]:   File "/home/ec2-user/axolotl/src/axolotl/cli/train.py", line 55, in do_train
[rank4]:     return train(cfg=cfg, cli_args=cli_args, dataset_meta=dataset_meta)
[rank4]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank4]:   File "/home/ec2-user/axolotl/src/axolotl/train.py", line 170, in train
[rank4]:     trainer.train(resume_from_checkpoint=resume_from_checkpoint)
[rank4]:   File "/home/ec2-user/micromamba/envs/axolotl/lib/python3.11/site-packages/transformers/trainer.py", line 1837, in train
[rank4]:     return inner_training_loop(
[rank4]:            ^^^^^^^^^^^^^^^^^^^^
[rank4]:   File "/home/ec2-user/micromamba/envs/axolotl/lib/python3.11/site-packages/transformers/trainer.py", line 2276, in _inner_training_loop
[rank4]:     self._maybe_log_save_evaluate(tr_loss, grad_norm, model, trial, epoch, ignore_keys_for_eval)
[rank4]:   File "/home/ec2-user/micromamba/envs/axolotl/lib/python3.11/site-packages/transformers/trainer.py", line 2651, in _maybe_log_save_evaluate
[rank4]:     self._save_checkpoint(model, trial, metrics=metrics)
[rank4]:   File "/home/ec2-user/micromamba/envs/axolotl/lib/python3.11/site-packages/transformers/trainer.py", line 2734, in _save_checkpoint
[rank4]:     self._save_optimizer_and_scheduler(output_dir)
[rank4]:   File "/home/ec2-user/micromamba/envs/axolotl/lib/python3.11/site-packages/transformers/trainer.py", line 2831, in _save_optimizer_and_scheduler
[rank4]:     self.model_wrapped.save_checkpoint(output_dir)
[rank4]:   File "/home/ec2-user/micromamba/envs/axolotl/lib/python3.11/site-packages/deepspeed/runtime/engine.py", line 3123, in save_checkpoint
[rank4]:     copytree(offload_dir,
[rank4]:   File "/home/ec2-user/micromamba/envs/axolotl/lib/python3.11/shutil.py", line 573, in copytree
[rank4]:     return _copytree(entries=entries, src=src, dst=dst, symlinks=symlinks,
[rank4]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank4]:   File "/home/ec2-user/micromamba/envs/axolotl/lib/python3.11/shutil.py", line 471, in _copytree
[rank4]:     os.makedirs(dst, exist_ok=dirs_exist_ok)
[rank4]:   File "<frozen os>", line 225, in makedirs
[rank4]: FileExistsError: [Errno 17] File exists: '/opt/dlami/nvme/fft-8b/checkpoint-9/global_step9/offloaded_tensors'
```

### Steps to reproduce

accelerate launch -m axolotl.cli.train fft-8b.yaml --deepspeed deepspeed/zero3_optim_nvme.json --output_dir /opt/dlami/nvme/fft-8b

Deepspeed config
```
{
  "zero_optimization": {
    "stage": 3,
    "overlap_comm": true,
    "contiguous_gradients": true,
    "sub_group_size": 0,
    "reduce_bucket_size": "auto",
    "stage3_prefetch_bucket_size": "auto",
    "stage3_param_persistence_threshold": "auto",
    "stage3_max_live_parameters": 0,
    "stage3_max_reuse_distance": 0,
    "stage3_gather_16bit_weights_on_model_save": true,
    "offload_optimizer": {
      "device": "nvme",
      "nvme_path": "/opt/dlami/nvme"
    }
  },
  "bf16": {
    "enabled": true
  },
  "fp16": {
    "enabled": "auto",
    "auto_cast": false,
    "loss_scale": 0,
    "initial_scale_power": 32,
    "loss_scale_window": 1000,
    "hysteresis": 2,
    "min_loss_scale": 1
  },
  "gradient_accumulation_steps": "auto",
  "gradient_clipping": "auto",
  "train_batch_size": "auto",
  "train_micro_batch_size_per_gpu": "auto",
  "wall_clock_breakdown": false
}
```

### Config yaml

```yaml
base_model: meta-llama/Meta-Llama-3-8B
model_type: LlamaForCausalLM
tokenizer_type: AutoTokenizer

load_in_8bit: false
load_in_4bit: false
strict: false

datasets:
  - path: tatsu-lab/alpaca
    type: alpaca
dataset_prepared_path: last_run_prepared
val_set_size: 0.01
output_dir:

sequence_len: 8192
sample_packing: true
pad_to_sequence_len: true

wandb_project:
wandb_entity:
wandb_watch:
wandb_name:
wandb_log_model:

gradient_accumulation_steps: 8
micro_batch_size: 1
num_epochs: 1
optimizer: adamw_torch 
lr_scheduler: cosine
learning_rate: 2e-5

train_on_inputs: false
group_by_length: false
bf16: true
fp16:
tf32: false

gradient_checkpointing: true
gradient_checkpointing_kwargs:
  use_reentrant: false
early_stopping_patience:
resume_from_checkpoint:
logging_steps: 1
xformers_attention:
flash_attention: true

warmup_steps: 100
evals_per_epoch: 2
eval_sample_packing: False
eval_table_size:
saves_per_epoch: 1
debug:
deepspeed:
weight_decay: 0.0
fsdp:
fsdp_config:
special_tokens:
  pad_token: <|end_of_text|>
```

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.11.9

### axolotl branch-commit

main/fff06af8d0514d41a62ce7a15ac17353b3b39fce

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by mariokostelac** [comment]
I can confirm that adding `save_strategy: "no"` successfully saves the model and process exits gracefully.

**Comment by winglian** [comment]
I believe this is similar to #1642 and resolved by #1643

**Comment by winglian** [comment]
Fixed by #1643

**Comment by mariokostelac** [comment]
Thanks @winglian and sorry if i duplicated the issue. I was trying to find the same one, but had no luck with issue search.
